File: DSA/SuryanshMishra_590012069/Day5/day5question2.cpp
#include<iostream>
#include<vector>
#include <unordered_map>
using namespace std;
void missingNo(vector<int>&arr, int n){
 int  curr_sum=0;
 int size_sum= n*(n+1)/2;
 for(int i=0;i<n;i++){
     curr_sum+=arr[i];
 }
 int missingNo= size_sum-curr_sum;
 if(missingNo) cout<<"The missing number in the array is:"<<" "<<missingNo;   
}
int main(){
    int n;
    cout<<"Enter the no of elements:"<<" ";
    cin>>n;
    vector<int>arr(n);
    cout<<"Enter the array elements: "<< " ";
    for(int i=0;i<n;i++) cin>>arr[i];
    missingNo(arr,n);
    
}

File: DSA/SuryanshShah_590013441/Day5Question2.java
import java.util.*;
public class Day5Question2 {
    public static void main(String[] args) {
        Scanner scr = new Scanner(System.in);
        System.out.println("Enter the number of elements in the array (should be n, where array has numbers from 0 to n):");
        int n = scr.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter the elements of the array (from 0 to n with one missing):");
        for (int i = 0; i < n; i++) {
            arr[i] = scr.nextInt();
        }
        Arrays.sort(arr);
        int missing = n; 
        for (int i = 0; i < n; i++) {
            if (arr[i] != i) {
                missing = i;
                break;
            }
        }
        System.out.println("The missing number is: " + missing);
    }
}
